src/Cli/dotnet/commands/dotnet-test/VSTestForwardingApp.cs

```diff
@@ -23,6 +23,15 @@ public VSTestForwardingApp(IEnumerable<string> argsToForward)
 
         private static string GetVSTestExePath()
         {
+            // Provide custom path to vstest.console.dll or exe to be able to test it against any version of 
+            // vstest.console. This is useful especially for our integration tests.
+            // This is equivalent to specifying -p:VSTestConsolePath when using dotnet test with csproj.
+            string vsTestConsolePath = Environment.GetEnvironmentVariable("VSTEST_CONSOLE_PATH");
+            if (!string.IsNullOrWhiteSpace(vsTestConsolePath))
```

Comment:
What if the path provided here is not correct or valid, for instance, the path does not exist. Is it validated somewhere?

Reply:
Not really, it will blow up later with file not found error when we try to run it. But since this is user setting it's their fault. And I don't want to limit my options of what I can set into the option.